packages/generator/src/edmx-to-vdm/common/action-function-return-types.ts

```diff
@@ -150,15 +150,26 @@ function getEdmReturnType(
 
 function getEntityReturnType(
   isCollection: boolean,
-  entity: VdmEntity
+  entities: VdmEntity[]
 ): VdmFunctionImportReturnType {
-  return {
-    returnTypeCategory: VdmReturnTypeCategory.ENTITY,
-    returnType: entity.className,
-    builderFunction: entity.className,
-    isMulti: isCollection,
-    isCollection
-  };
+  if (!entities.length) {
+    throw Error('Cannot get entity return type from an empty collection.');
```

Comment:
"collection" might be misleading here. Consider something in this direction: `Could not get entity return type for function import. No matching entity types found.`